docs/datasets.md

```diff
@@ -0,0 +1,212 @@
+description. The `id` of the created dataset is in JSON data of the response.
+
+```python
+>>> datasets = Datasets()
+>>> create_resp = datasets.create(
+...     name='example', description='An example dataset')
+>>> new = create_resp.json()
+>>> new['name']
+'example'
+>>> new['description']
+'An example dataset'
+>>> new_id = new['id']
+
+```
+
+You can find it in your account's list of datasets.
+
+```python
+>>> listing_resp = datasets.list()
+>>> new_id in [ds['id'] for ds in listing_resp.json()]
+True
+
+```
+
+Instead of scanning the list for attributes of the dataset, you can read
+them directly by dataset id.
+
+```python
+>>> attrs = datasets.read_dataset(new_id).json()
+>>> attrs['id'] == new_id
+True
+>>> attrs['name']
+'example'
+>>> attrs['description']
+'An example dataset'
+
+```
+
+If you want to change a dataset's name or description, you can.
+
+```python
+>>> _ = datasets.update_dataset(
+...     new_id, name='updated example', description='An updated example dataset'
+...     ).json()
+>>> attrs = datasets.read_dataset(new_id).json()
+>>> attrs['id'] == new_id
+True
+>>> attrs['name']
+'updated example'
+>>> attrs['description']
+'An updated example dataset'
+
+```
+
+You can delete the dataset and it will no longer be present in your listing.
+
+```python
+>>> resp = datasets.delete_dataset(new_id)
+>>> resp.status_code
+204
+>>> listing_resp = datasets.list()
+>>> new_id in [ds['id'] for ds in listing_resp.json()]
+False
+
+```
+
+## Dataset features
+
+The main point of a dataset is store a collection of GeoJSON features. Let us
+create a new dataset and then add a GeoJSON feature to it.
+
+```python
+>>> resp = datasets.create(
+...     name='features-example', description='An example dataset with features')
+>>> new_id = resp.json()['id']
+>>> feature = {
+...     'type': 'Feature', 'id': '1', 'properties': {'name': 'Insula Nulla'},
+...     'geometry': {'type': 'Point', 'coordinates': [0, 0]}}
+>>> resp = datasets.batch_update_features(new_id, put=[feature])
+
+```
+
+In the feature collection of the dataset you will see this feature.
+
+```python
+>>> collection = datasets.list_features(new_id).json()
+>>> len(collection['features'])
+1
+>>> first = collection['features'][0]
+>>> first['id']
+'1'
+>>> first['properties']['name']
+'Insula Nulla'
+
+```
+
+Using the same `batch_update_features()` method, you can modify or delete
+up to 100 features.
+
+```python
+>>> feature2 = {
+...     'type': 'Feature', 'id': '2', 'properties': {'name': 'Insula Nulla B'},
+...     'geometry': {'type': 'Point', 'coordinates': [0, 0]}}
+>>> resp = datasets.batch_update_features(new_id, put=[feature2], delete=['1'])
+>>> resp.status_code
+200
+
+```
+
+The results:
+
+```python
+
+>>> updated_collection = datasets.list_features(new_id).json()
+>>> len(updated_collection['features'])
+1
+>>> updated_first = updated_collection['features'][0]
+>>> updated_first['id'] # should return '2'
```

Comment:
if this is an eventual consistency issue, isn't it also possible that `list_features` could return feature `2` under certain circumstances? Maybe we should just note it but not test it until we find a way to poll the service.